### PR TITLE
Scripting: capture structured javadoc from stdlib

### PR DESCRIPTION
**Pull request by stu-elastic**
Clean javadoc tags and strip html.
Methods and constructors have an optional `javadoc` field.  All fields under
`javadoc` are optional but at least one will be present.

Fields also have optional `javadoc` field which, if present, is a string.
```
"javadoc": {
  "description": "...",

  // from @param <param name> <param description>
  "parameters": {
    "p1": "<p1 description>",
    "p2": "<p2 description>"
  },

  // from @return
  "return": "...",

  // from @throws <type> <description>
  "throws": [
    [
      "IndexOutOfBoundsException",
      "<description>"
    ],
    [
      "IOException",
      "<description>"
    ]
  ]
}
```

### CODE REVIEW

File: modules/lang-painless/build.gradle [hunk 1]
```diff
@@ -77,13 +77,15 @@ dependencies {
   docImplementation project(':server')
   docImplementation project(':modules:lang-painless')
   docImplementation 'com.github.javaparser:javaparser-core:3.18.0'
+  docImplementation 'org.jsoup:jsoup:1.13.1'
```
Comment (jdconrad) [inline modules/lang-painless/build.gradle:80]: What's this do?

File: modules/lang-painless/src/doc/java/org/elasticsearch/painless/JavadocExtractor.java [hunk 2]
```diff
@@ -192,15 +202,103 @@ public int hashCode() {
...
 
+    public static ParsedJavadoc clean(Javadoc javadoc) {
+        JavadocDescription description = javadoc.getDescription();
+        List<JavadocBlockTag> tags = javadoc.getBlockTags();
+        List<String> cleaned = new ArrayList<>(description.getElements().size() + tags.size());
+        cleaned.addAll(stripInlineTags(description));
+        ParsedJavadoc parsed = new ParsedJavadoc(cleaned(cleaned));
+        for (JavadocBlockTag tag: tags) {
+            String tagName = tag.getTagName();
+            // https://docs.oracle.com/javase/7/docs/technotes/tools/windows/javadoc.html#javadoctags
```
Comment (jdconrad) [inline modules/lang-painless/src/doc/java/org/elasticsearch/painless/JavadocExtractor.java:222]: Did you intend to link to Java 7 for this comment?

**Review by jdconrad — APPROVED** [review]
LGTM! Nice update.

File: modules/lang-painless/build.gradle (hunk 1 above)
Comment (stu-elastic) [reply modules/lang-painless/build.gradle:80]: It's used to [clean html tags](https://github.com/jhy/jsoup/).

File: modules/lang-painless/src/doc/java/org/elasticsearch/painless/JavadocExtractor.java (hunk 2 above)
Comment (stu-elastic) [reply modules/lang-painless/src/doc/java/org/elasticsearch/painless/JavadocExtractor.java]: Updated to java14.

**Pull request merged**